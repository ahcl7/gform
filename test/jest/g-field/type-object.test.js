import { generateComponent } from "./common";

describe('GField with type object', function () {
  it('should render correctly', () => {
    const component = generateComponent({
      key: 'address', type: 'object',
      fields: [{ key: 'street', type: 'input', flex: 'md-6' }, { key: 'city', type: 'input', flex: 'md-6' },]
    });
    expect(component.html()).toMatchSnapshot();
  });
});
