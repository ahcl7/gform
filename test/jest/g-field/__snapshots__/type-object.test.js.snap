// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GField with type object should render correctly 1`] = `
"<div class=\\"g-col col-xs-12\\" xs12=\\"\\">
  <fieldset class=\\"\\" style=\\"position: relative;\\">
    <!---->
    <div style=\\"display: none;\\">
      <div class=\\"action-container g-card bg-white\\">
        <div class=\\"g-btn g-btn__flat g-btn__icon g-btn__round g-btn__uppercase g-size__x-small waves-effect\\"><span class=\\"g-btn__content\\"><i class=\\"g-icon material-icons\\" style=\\"font-size: 16px;\\">delete</i></span></div>
      </div>
    </div>
    <legend class=\\"\\"><span>Address </span></legend>
    <transition-stub name=\\"expand-transition\\" mode=\\"in-out\\">
      <div class=\\"g-row\\">
        <div class=\\"g-row no-gutters\\">
          <div class=\\"g-col md-6 px-2\\">
            <div class=\\"g-tf-wrapper g-tf r\\">
              <!---->
              <!----> <input autocomplete=\\"off\\" label=\\"Street\\" type=\\"text\\" class=\\"g-tf-input\\">
              <!----> <label class=\\"g-tf-label\\">
                Street
                <!----></label>
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"g-col md-6 px-2\\">
            <div class=\\"g-tf-wrapper g-tf r\\">
              <!---->
              <!----> <input autocomplete=\\"off\\" label=\\"City\\" type=\\"text\\" class=\\"g-tf-input\\">
              <!----> <label class=\\"g-tf-label\\">
                City
                <!----></label>
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          <div class=\\"g-col col-xs-12\\" xs12=\\"\\"></div>
        </div>
      </div>
    </transition-stub>
  </fieldset>
</div>"
`;
